packages/core/src/EventNormalizer.ts

```diff
@@ -146,326 +536,812 @@ export class EventNormalizer {
+
+        const googleKeyboardKey =
+            keydownEvent &&
+            keydownEvent.key === 'Unidentified' &&
+            inputEvent &&
+            inputEvent.inputType === 'insertCompositionText' &&
+            compositionStartEvent &&
+            compositionStartEvent.data === '' &&
+            this._events.filter(event => event.type === 'compositionstart').length === 1 &&
+            this._events.filter(event => event.type === 'compositionupdate').length === 2 &&
+            this._events.filter(event => event.type === 'compositionend').length === 0 &&
+            inputEvent.data[inputEvent.data.length - 1];
+        const swiftKeyInsertKey =
+            keydownEvent &&
+            keydownEvent.key === 'Unidentified' &&
+            inputEvent &&
+            inputEvent.inputType === 'insertText' &&
+            !googleKeyboardKey &&
+            this._events.filter(event => event.type === 'input').length === 1 &&
+            inputEvent.data;
+
+        //
+        // First pass to get the informations
+        //
+        const key =
+            (keypressEvent &&
+                keypressEvent.key !== 'Unidentified' &&
+                keypressEvent.key !== 'Dead' &&
+                keypressEvent.key) ||
+            (inputEvent &&
+                inputEvent.data !== null &&
+                inputEvent.data.length === 1 &&
+                inputEvent.data) ||
+            (keydownEvent &&
+                keydownEvent.key !== 'Unidentified' &&
+                keydownEvent.key !== 'Dead' &&
+                keydownEvent.key) ||
+            (isGoogleKeyboardBackspace && 'Backspace') ||
+            googleKeyboardKey ||
+            swiftKeyInsertKey ||
+            (keydownEvent &&
+                keydownEvent.key === 'Unidentified' &&
+                this._inferKeyFromInput(inputEvent));
+        // in the test "accent (mac safari)", an erroneous code is triggered
+        // (BracketLeft).  to prevent the code from being set, we check if a key
+        // is dead.
+        const isDeadKey = keydownEvent && keydownEvent.key === 'Dead';
+        const code =
+            !isDeadKey &&
+            ((keydownEvent && keydownEvent.code) || (keypressEvent && keypressEvent.code));
+        const isAccentMac =
+            inputEvent &&
+            compositionEvent &&
+            compositionEvent.data &&
+            compositionEvent.data.length > 0;
+        const inputType =
+            (cutEvent && 'deleteByCut') ||
+            (dropEvent && 'insertFromDrop') ||
+            (pasteEvent && 'insertFromPaste') ||
+            (isGoogleKeyboardBackspace && 'deleteContentBackward') ||
+            (!!googleKeyboardKey && 'insertText') ||
+            (inputEvent && inputEvent.inputType) ||
+            // todo: Do we really need to set the inputType when making a
+            //       "special accent" in mac?
+            (isAccentMac && 'insertCompositionText');
+        const defaultPrevented =
```

Comment:
never used. Why ? It's important to allow the prevent default by plugins and external libs